### DISCUSSION TITLE
how to use table-block in angular

### DISCUSSION THREAD

**Discussion by solos** [Milkdown/milkdown#1539] [Q&A] [answered]
When i import the table-block as https://milkdown.dev/docs/api/component-table-block , it failed to compile. 
Could i use the table-block in angular project?

```
Error: node_modules/@milkdown/components/lib/table-block/view/component.d.ts:14:36 - error TS2314: Generic type 'Atomico<Props, PropsForInstance, Base>' requires 3 type argument(s).

export declare const TableElement: import("atomico/types/dom").Atomico<TableComponentProps | (TableComponentProps & import("atomico/types/component").SyntheticMetaProps<any>), {
                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  new (): HTMLElement;
   ~~~~~~~~~~~~~~~~~~~~~~~~
   prototype: HTMLElement;
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~
 }>;
   ~~
```

![image](https://github.com/user-attachments/assets/f0ad5ef8-1c44-44f7-854c-1e7481906655)

**Comment by Saul-Mirone** [comment]
Do you have a minimal reproduction? It should work in all frameworks.

**Reply by solos** [reply]
Yes, here is the demo. When i import tableBlock, it will fail to compile. When i remove the import , it will work. 
import { tableBlock } from '@milkdown/kit/component/table-block';

 https://stackblitz.com/edit/stackblitz-starters-gppxafrp?file=src%2Fapp%2Fmain%2Feditor%2Feditor.component.ts

**Reply by solos** [reply, marked as answer]
Sorry for the issue, when i use  atomico 1.75.1 version, it will be ok. I am not using the same version package.
